src/app/beer_garden/router.py

```diff
@@ -154,127 +183,182 @@ def forward(operation: Operation):
         RoutingRequestException: Could not determine a route to child
         UnknownGardenException: The specified target garden is unknown
     """
-    target_garden = None
-    for garden in gardens:
-        if garden.name == operation.target_garden_name:
-            target_garden = garden
-            break
-
-    if not target_garden:
+    try:
+        conn_type, conn_info = garden_connections[operation.target_garden_name]
+    except KeyError:
```

Comment:
Still reading through, but would it be an error if `conn_type` and `conn_info` are `None`? If the admin has not configured the garden, then that will happen. We could raise an exception saying that Garden is not configured, please talk to your admins.

Reply:
I added a check for this case.

Reply:
:+1: